packages/vscode-container-client/src/test/ContainersClientE2E.test.ts

```diff
@@ -0,0 +1,1007 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { expect } from 'chai';
+import * as fs from 'fs/promises';
+import * as os from 'os';
+import * as path from 'path';
+import * as stream from 'stream';
+import { FileType } from 'vscode';
+import { DockerClient } from '../clients/DockerClient/DockerClient';
+import { PodmanClient } from '../clients/PodmanClient/PodmanClient';
+import { ShellStreamCommandRunnerFactory, ShellStreamCommandRunnerOptions } from '../commandRunners/shellStream';
+import { WslShellCommandRunnerFactory, WslShellCommandRunnerOptions } from '../commandRunners/wslStream';
+import { IContainersClient, ListContainersItem, ListImagesItem, ListNetworkItem, ListVolumeItem, StatPathItem } from '../contracts/ContainerClient';
+import { CommandResponseBase, ICommandRunnerFactory } from '../contracts/CommandRunner';
+import { isCommandNotSupportedError } from '../utils/CommandNotSupportedError';
+import { Bash } from '../utils/spawnStreamAsync';
+import { wslifyPath } from '../utils/wslifyPath';
+
+/**
+ * WARNING: This test suite will prune unused images, containers, networks, and volumes.
+ */
+
+// Modify the below options to configure the tests
+const clientTypeToTest: ClientType = 'docker' as ClientType;
+const runInWsl: boolean = false; // Set to true if running in WSL
+
+// Supply to run the login/logout tests
+const dockerHubUsername = '';
+const dockerHubPAT = ''; // Never commit this value!!

```

Comment:
Here I am also wondering if env vars would be better.

Reply:
Agreed